hazelcast/src/main/java/com/hazelcast/cp/internal/RaftService.java

```diff
@@ -593,10 +596,22 @@ public void memberRemoved(MembershipServiceEvent event) {
         updateMissingMembers();
     }
 
+
     private void publishGroupAvailabilityEvents(MemberImpl removedMember) {
-        ClusterService clusterService = nodeEngine.getClusterService();
+        publishGroupAvailabilityEvents(removedMember, false);
+    }
+
+    private void publishGroupAvailabilityEventsForGracefulShutdown(MemberImpl removedMember) {
+        // TODO CPGroupAvailabilityEventGracefulImpl added in V5_4; we require cluster to be on at least V5_4 for its publication
```

Comment:
use the convention for marking rolling-upgrade related code: `// RU_COMPAT_5_3` (indicating the version with which we are maintaining compatibility), so it is easier to find for clean up in next dev iteration.